libs/partners/postgres/langchain_postgres/checkpoint.py

```diff
@@ -0,0 +1,313 @@
+                            else None,
+                        )
+                else:
+                    await cur.execute(
+                        "SELECT checkpoint, thread_ts, parent_ts "
+                        "FROM checkpoints "
+                        "WHERE thread_id = %(thread_id)s "
+                        "ORDER BY thread_ts DESC LIMIT 1",
+                        {
+                            "thread_id": thread_id,
+                        },
+                    )
+                    value = await cur.fetchone()
+                    if value:
+                        return CheckpointTuple(
+                            config={
+                                "configurable": {
+                                    "thread_id": thread_id,
+                                    "thread_ts": value[1].isoformat(),
+                                }
+                            },
+                            checkpoint=pickle.loads(value[0]),
+                            parent_config={
+                                "configurable": {
+                                    "thread_id": thread_id,
+                                    "thread_ts": value[2].isoformat(),
+                                }
+                            }
+                            if value[2]
+                            else None,
+                        )
+
+    async def aput(
+        self, config: RunnableConfig, checkpoint: Checkpoint
+    ) -> RunnableConfig:
+        """Put the checkpoint for the given configuration.
+
+        Args:
+            config: The configuration for the checkpoint.
+                A dict with a `configurable` key which is a dict with
+                a `thread_id` key and an optional `thread_ts` key.
+                For example, { 'configurable': { 'thread_id': 'test_thread' } }
+            checkpoint: That was saved.
+
+        Returns:
+            The configuration for the checkpoint.
+        """
+        thread_id = config["configurable"]["thread_id"]
+        parent_ts = config["configurable"].get("thread_ts")
+        async with self._get_async_connection() as conn:
+            async with conn.cursor() as cur:
+                await cur.execute(
+                    """
+                    INSERT INTO checkpoints (thread_id, thread_ts, parent_ts, checkpoint)
+                    VALUES (%(thread_id)s, %(thread_ts)s, %(parent_ts)s, %(checkpoint)s)
+                    ON CONFLICT (thread_id, thread_ts) 
+                    DO UPDATE SET checkpoint = EXCLUDED.checkpoint;
+                    """,
+                    {
+                        "thread_id": thread_id,
+                        "thread_ts": checkpoint["ts"],
+                        "parent_ts": parent_ts if parent_ts else None,
+                        "checkpoint": pickle.dumps(checkpoint),
+                    },
+                )
+
+        return {
+            **config,  # TODO(Nuno): Confirm
```

Comment:
@nfcampos need review here -- unclear what is meant to be returned by this method or why

Reply:
this is the config that uniquely identifies the checkpoint that was just saved (the primary key)